scripts/packages/VSCodeServer/src/eval.jl

```diff
@@ -174,13 +175,30 @@ function repl_runcode_request(conn, params::ReplRunCodeRequestParams)::ReplRunCo
                     if show_code
                         REPL_PROMPT_STATE[] = REPLPromptStates.Error
                     end
-                    @static if isdefined(Base, :current_exceptions)
+                    errs = @static if isdefined(Base, :current_exceptions)
                         EvalErrorStack(Base.current_exceptions(current_task()))
                     elseif isdefined(Base, :catch_stack)
                         EvalErrorStack(Base.catch_stack())
                     else
                         EvalError(err, catch_backtrace())
                     end
+                    if CAN_SET_ERR[]
+                        try
+                            istrivial = @static if isdefined(Base, :istrivialerror)
+                                Base.istrivialerror(errs.stack)
+                            else
+                                true
+                            end
+                            @static if @isdefined setglobal!
+                                istrivial || setglobal!(Main, :err, errs.stack)
+                            else
+                                istrivial || ccall(:jl_set_global, Cvoid, (Any, Any, Any), Main, :err, errs.stack)
+                            end
```

Comment:
Wanna put in some additional safe guards here like `errs isa EvalErrorStack`? I realize that's  never going to be false because `istrivialerror` was introduced later, but still :)

Reply:
Let me know if that change makes sense - since the design is "try once, if it throws an exception that's okay, we just won't try anymore", throwing an error seemed easiest.